File: Javascript/7kyu/reverse-words.js
//Describtion:
// Complete the function that accepts a string parameter, and reverses each word in the string. All spaces in the string should be retained.
// Examples

// "This is an example!" ==> "sihT si na !elpmaxe"
// "double  spaces"      ==> "elbuod  secaps"


//My Solution:
function reverseWords(str) {
    return str.split(' ').map(word => word.split('').reverse().join('')).join(' ');
   }

//Other solutions:
function reverseWords(str) {
    // Go for it
    //split words into seperate arrays
    return str.split("").reverse().join("").split(" ").reverse().join(" ");
  }

File: Python/7kyu/reverse-words.py
#Describtion:
# Complete the function that accepts a string parameter, and reverses each word in the string. All spaces in the string should be retained.
# Examples

# "This is an example!" ==> "sihT si na !elpmaxe"
# "double  spaces"      ==> "elbuod  secaps"

#Solution:
def reverse_words(text):
    return ' '.join(s[::-1] for s in text.split(' '))

#Other solutions:
def reverse_words(str):
  #go for it
  newStr = []
  for i in str.split(' '):
      newStr.append(i[::-1])
  return ' '.join(newStr)